File: libs/domain/models-rs/src/finding.rs
// libs/domain/models-rs/src/finding.rs
// =================================================================
// APARATO: FINDING DOMAIN MODEL (V10.0)
// RESPONSABILIDAD: DEFINICIÓN DE COLISIÓN CRIPTOGRÁFICA
// ESTADO: NO-REGRESSIONS / AUDIT-READY
// =================================================================

use serde::{Deserialize, Serialize};
use typeshare::typeshare;

/// El "Santo Grial". Representa una colisión exitosa en el espacio secp256k1.
#[typeshare]
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Finding {
    /// La dirección Bitcoin (P2PKH) que contenía balance.
    pub address: String,

    /// La clave privada en formato WIF.
    /// En tránsito entre Worker y Orquestador, este campo es sensible.
    pub private_key_wif: String,

    /// Descripción de la fuente de entropía (ej: "dictionary:rockyou:line_450").
    pub source_entropy: String,

    /// Tipo de dirección: 'legacy_uncompressed', 'legacy_compressed'.
    pub wallet_type: String,

    /// --- METADATOS DE AUDITORÍA ---

    /// ID del worker que reportó el hallazgo.
    pub found_by_worker: String,

    /// ID del trabajo (rango) donde ocurrió la colisión.
    pub job_id: Option<String>,

    /// Timestamp UTC de la detección.
    pub detected_at: String,
}

File: libs/domain/api-contracts/src/lib/finding.ts
/**
 * =================================================================
 * APARATO: FINDING DATA CONTRACT
 * CLASIFICACIÓN: DOMAIN CONTRACTS (L2)
 * RESPONSABILIDAD: DEFINICIÓN DE COLISIÓN CRIPTOGRÁFICA
 * ESTADO: ELITE TYPING (ZK-COMPLIANT)
 * =================================================================
 */

import { z } from "zod";

/**
 * Esquema de validación para una colisión confirmada.
 * Representa el material criptográfico recuperado del set UTXO.
 */
export const FindingSchema = z.object({
  /** Dirección Bitcoin (P2PKH) colisionada */
  address: z.string().min(26).max(35),

  /** Clave privada en formato WIF (Wallet Import Format) */
  private_key_wif: z.string(),

  /** Metadatos de la fuente de entropía (ej: 'brainwallet:pass123') */
  source_entropy: z.string(),

  /** Clasificación de la billetera: legacy_uncompressed, legacy_compressed */
  wallet_type: z.string(),

  /** Identificador único del nodo que realizó el hallazgo */
  found_by_worker: z.string(),

  /** Identificador de la orden de trabajo (rango) origen */
  job_id: z.string().uuid().nullable().optional(),

  /** Marca de tiempo de la detección (ISO 8601 UTC) */
  detected_at: z.string().datetime(),
});

/**
 * Tipo inferido para uso en el enjambre y el orquestador.
 */
export type Finding = z.infer<typeof FindingSchema>;
